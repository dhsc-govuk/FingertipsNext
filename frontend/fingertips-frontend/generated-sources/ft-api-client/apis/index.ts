/* tslint:disable */
/* eslint-disable */
export * from './AreasApi';
export * from './BatchesApi';
export * from './IndicatorsApi';
export * from './SystemApi';
export * from './UserApi';
