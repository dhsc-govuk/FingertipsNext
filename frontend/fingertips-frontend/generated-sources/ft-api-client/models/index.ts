/* tslint:disable */
/* eslint-disable */
export * from './AgeData';
export * from './Area';
export * from './AreaType';
export * from './AreaWithRelations';
export * from './BenchmarkComparisonMethod';
export * from './BenchmarkOutcome';
export * from './BenchmarkReferenceType';
export * from './DeprivationData';
export * from './GetAreaHierarchies500Response';
export * from './GetHealthcheck200Response';
export * from './GetHealthcheck500Response';
export * from './HealthDataForArea';
export * from './HealthDataPoint';
export * from './HealthDataPointBenchmarkComparison';
export * from './Indicator';
export * from './IndicatorPolarity';
export * from './IndicatorSummary';
export * from './IndicatorWithHealthDataForArea';
export * from './PeriodType';
export * from './QuartileData';
export * from './RootArea';
export * from './SexData';
