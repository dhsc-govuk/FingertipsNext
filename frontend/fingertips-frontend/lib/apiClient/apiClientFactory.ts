import {
  AreasApi,
  BatchesApi,
  Configuration,
  IndicatorsApi,
  SystemApi,
  UserApi,
} from '@/generated-sources/ft-api-client';
import { readEnvVar } from '../envUtils';
import { getAuthHeader } from '@/lib/auth/accessToken';

export const API_CACHE_CONFIG = { next: { revalidate: 600 } };

const buildConfig = (): Configuration => {
  const apiUrl = readEnvVar('FINGERTIPS_API_URL');
  return new Configuration({
    basePath: apiUrl,
    fetchApi: fetch,
  });
};

const buildConfigWithAuthHeader = async (): Promise<Configuration> => {
  const apiUrl = readEnvVar('FINGERTIPS_API_URL');
  return new Configuration({
    basePath: apiUrl,
    fetchApi: fetch,
    headers: await getAuthHeader(),
  });
};

export class ApiClientFactory {
  private static areasApiInstance: AreasApi | null;
  private static indicatorsApiInstance: IndicatorsApi | null;
  private static systemApiInstance: SystemApi | null;
  private static userApiInstance: UserApi | null;
  private static batchesApiInstance: BatchesApi | null;

  public static getAreasApiClient(): AreasApi {
    if (!this.areasApiInstance) {
      this.areasApiInstance = new AreasApi(buildConfig());
    }

    return this.areasApiInstance;
  }

  public static getIndicatorsApiClient(): IndicatorsApi {
    if (!this.indicatorsApiInstance) {
      this.indicatorsApiInstance = new IndicatorsApi(buildConfig());
    }

    return this.indicatorsApiInstance;
  }

  public static async getAuthenticatedIndicatorsApiClient(): Promise<IndicatorsApi> {
    return new IndicatorsApi(await buildConfigWithAuthHeader());
  }

  public static getSystemApiClient(): SystemApi {
    if (!this.systemApiInstance) {
      this.systemApiInstance = new SystemApi(buildConfig());
    }

    return this.systemApiInstance;
  }

  public static getUserApiClient(): UserApi {
    if (!this.userApiInstance) {
      this.userApiInstance = new UserApi(buildConfig());
    }

    return this.userApiInstance;
  }

  public static getBatchesApiClient(): BatchesApi {
    if (!this.batchesApiInstance) {
      this.batchesApiInstance = new BatchesApi(buildConfig());
    }

    return this.batchesApiInstance;
  }

  public static async getAuthenticatedBatchesApiClient(): Promise<BatchesApi> {
    return new BatchesApi(await buildConfigWithAuthHeader());
  }
}
