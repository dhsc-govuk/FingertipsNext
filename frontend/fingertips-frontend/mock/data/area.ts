export const mockAreaData = (id: string) => ({
  id,
  name: `Greater Manchester ICB - ${id} sub-location`,
  groupType: `GT-${id}`,
  group: 'QOP',
});

export const mockAvailableGroupTypes = [
  {
    id: 'GT-001',
    name: 'Integrated Care Board',
  },
  {
    id: 'GT-002',
    name: 'Integrated Care Panel',
  },
  {
    id: 'GT-003',
    name: 'Integrated Care Shelf',
  },
];

export const mockAvailableGroup = [
  {
    id: 'QOP',
    name: 'Greater Manchester  ICB - QOP',
  },
  {
    id: 'LOP',
    name: 'Leeds  ICB - LOP',
  },
  {
    id: 'SOP',
    name: 'Sheffield  ICB - SOP',
  },
];

export const mockAvailableAreasInGroup = [
  {
    id: '001',
    name: 'Greater Manchester ICN - 001',
  },
  {
    id: '002',
    name: 'Greater Manchester ICN - 002',
  },
  {
    id: '003',
    name: 'Greater Manchester ICN - 003',
  },
  {
    id: '004',
    name: 'Greater Manchester ICN - 004',
  },
  {
    id: '005',
    name: 'Greater Manchester ICN - 005',
  },
  {
    id: '006',
    name: 'Greater Manchester ICN - 006',
  },
  {
    id: '007',
    name: 'Greater Manchester ICN - 007',
  },
  {
    id: '008',
    name: 'Greater Manchester ICN - 008',
  },
  {
    id: '009',
    name: 'Greater Manchester ICN - 008',
  },
  {
    id: '010',
    name: 'Greater Manchester ICN - 010',
  },
];
