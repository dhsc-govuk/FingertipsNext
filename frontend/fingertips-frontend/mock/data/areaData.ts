import { AreaType, AreaWithRelations } from '@/generated-sources/ft-api-client';

export const mockAreaTypes: AreaType[] = [
  {
    name: 'Counties & UAs',
    level: 2,
    hierarchyName: 'Admin',
  },
  {
    name: 'Country',
    level: 0,
    hierarchyName: 'All',
  },
  {
    name: 'GP',
    level: 4,
    hierarchyName: 'NHS',
  },
  {
    name: 'ICB',
    level: 2,
    hierarchyName: 'NHS',
  },
  {
    name: 'NHS region',
    level: 1,
    hierarchyName: 'NHS',
  },
  {
    name: 'PCN',
    level: 3,
    hierarchyName: 'NHS',
  },
  {
    name: 'Regions Statistical',
    level: 1,
    hierarchyName: 'Admin',
  },
];

export const mockAreaDataForCountry: Record<string, AreaWithRelations> = {
  E92000001: {
    code: 'E92000001',
    name: 'England',
    hierarchyName: 'All',
    areaType: 'Country',
    level: 0,
  },
};

export const mockAreaDataForRegionsStatistical: Record<
  string,
  AreaWithRelations
> = {
  E12000001: {
    code: 'E12000001',
    name: 'North East region (statistical)',
    hierarchyName: 'Admin',
    areaType: 'Regions Statistical',
    level: 1,
  },
  E12000002: {
    code: 'E12000002',
    name: 'North West region (statistical)',
    hierarchyName: 'Admin',
    areaType: 'Regions Statistical',
    level: 1,
  },
  E12000003: {
    code: 'E12000003',
    name: 'Yorkshire and the Humber region (statistical)',
    hierarchyName: 'Admin',
    areaType: 'Regions Statistical',
    level: 1,
  },
};

export const mockAreaDataForNHSRegion: Record<string, AreaWithRelations> = {
  E40000007: {
    code: 'E40000007',
    name: 'East of England NHS Region',
    hierarchyName: 'NHS',
    areaType: 'NHS region',
    level: 1,
  },
  E40000003: {
    code: 'E40000003',
    name: 'London NHS Region',
    hierarchyName: 'NHS',
    areaType: 'NHS region',
    level: 1,
  },
  E40000005: {
    code: 'E40000005',
    name: 'South East NHS Region',
    hierarchyName: 'NHS',
    areaType: 'NHS region',
    level: 1,
  },
  E40000006: {
    code: 'E40000006',
    name: 'South West NHS Region',
    hierarchyName: 'NHS',
    areaType: 'NHS region',
    level: 1,
  },
  E40000010: {
    code: 'E40000010',
    name: 'North West NHS Region',
    hierarchyName: 'NHS',
    areaType: 'NHS region',
    level: 1,
  },
  E40000011: {
    code: 'E40000011',
    name: 'Midlands NHS Region',
    hierarchyName: 'NHS',
    areaType: 'NHS region',
    level: 1,
  },
  E40000012: {
    code: 'E40000012',
    name: 'North East and Yorkshire NHS Region',
    hierarchyName: 'NHS',
    areaType: 'NHS region',
    level: 1,
  },
};

export const mockAreaDataForCountiesAndUAs: Record<string, AreaWithRelations> =
  {
    E06000047: {
      code: 'E06000047',
      name: 'County Durham',
      hierarchyName: 'Admin',
      areaType: 'Counties & UAs',
      level: 2,
    },
    E06000005: {
      code: 'E06000005',
      name: 'Darlington',
      hierarchyName: 'Admin',
      areaType: 'Counties & UAs',
      level: 2,
    },
    E08000037: {
      code: 'E08000037',
      name: 'Gateshead',
      hierarchyName: 'Admin',
      areaType: 'Counties & UAs',
      level: 2,
    },
  };

export const mockAreaDataForICB: Record<string, AreaWithRelations> = {
  E38000007: {
    code: 'E38000007',
    name: 'NHS Basildon And Brentwood ICB',
    hierarchyName: 'NHS',
    areaType: 'ICB',
    level: 2,
  },
  E38000026: {
    code: 'E38000026',
    name: 'NHS Cambridgeshire and Peterborough ICB',
    hierarchyName: 'NHS',
    areaType: 'ICB',
    level: 2,
  },
  E38000240: {
    code: 'E38000240',
    name: 'NHS North Central London ICB',
    hierarchyName: 'NHS',
    areaType: 'ICB',
    level: 2,
  },
  E38000244: {
    code: 'E38000244',
    name: 'NHS South East London ICB',
    hierarchyName: 'NHS',
    areaType: 'ICB',
    level: 2,
  },
};

export const mockAreaDataForPCN: Record<string, AreaWithRelations> = {
  U15488: {
    code: 'U15488',
    name: 'East Basildon PCN',
    hierarchyName: 'NHS',
    areaType: 'PCN',
    level: 3,
  },
  U55146: {
    code: 'U55146',
    name: 'Central Basildon PCN',
    hierarchyName: 'NHS',
    areaType: 'PCN',
    level: 3,
  },
  U02795: {
    code: 'U02795',
    name: 'North 2 Islington PCN',
    hierarchyName: 'NHS',
    areaType: 'PCN',
    level: 3,
  },
  U05885: {
    code: 'U05885',
    name: 'South Camden PCN',
    hierarchyName: 'NHS',
    areaType: 'PCN',
    level: 3,
  },
};

export const mockAreaDataForGP: Record<string, AreaWithRelations> = {
  F81186: {
    code: 'F81186',
    name: 'Felmores Medical Centre',
    hierarchyName: 'NHS',
    areaType: 'GP',
    level: 4,
  },
  F81640: {
    code: 'F81640',
    name: 'Aryan Medical Centre',
    hierarchyName: 'NHS',
    areaType: 'GP',
    level: 4,
  },
  F83004: {
    code: 'F83004',
    name: 'Archway Medical Centre',
    hierarchyName: 'NHS',
    areaType: 'GP',
    level: 4,
  },
  F83008: {
    code: 'F83008',
    name: 'The Goodinge Group Practice',
    hierarchyName: 'NHS',
    areaType: 'GP',
    level: 4,
  },
};

export const mockAreaData: Record<string, AreaWithRelations> = {
  ...mockAreaDataForCountry,
  ...mockAreaDataForNHSRegion,
  ...mockAreaDataForRegionsStatistical,
  ...mockAreaDataForCountiesAndUAs,
  ...mockAreaDataForICB,
  ...mockAreaDataForPCN,
  ...mockAreaDataForGP,
};

export const mockAvailableAreas: Record<string, Area[]> = {
  'Counties & UAs': [
    {
      code: 'E06000047',
      name: 'County Durham',
      hierarchyName: 'Admin',
      areaType: 'Counties & UAs',
      level: 2,
    },
    {
      code: 'E06000005',
      name: 'Darlington',
      hierarchyName: 'Admin',
      areaType: 'Counties & UAs',
      level: 2,
    },
    {
      code: 'E08000037',
      name: 'Gateshead',
      hierarchyName: 'Admin',
      areaType: 'Counties & UAs',
      level: 2,
    },
  ],
  'Country': [
    {
      code: 'E92000001',
      name: 'England',
      hierarchyName: 'All',
      areaType: 'Country',
      level: 0,
    },
  ],
  'GP': [
    {
      code: 'F81186',
      name: 'Felmores Medical Centre',
      hierarchyName: 'NHS',
      areaType: 'GP',
      level: 4,
    },
    {
      code: 'F81640',
      name: 'Aryan Medical Centre',
      hierarchyName: 'NHS',
      areaType: 'GP',
      level: 4,
    },
    {
      code: 'F83004',
      name: 'Archway Medical Centre',
      hierarchyName: 'NHS',
      areaType: 'GP',
      level: 4,
    },
    {
      code: 'F83008',
      name: 'The Goodinge Group Practice',
      hierarchyName: 'NHS',
      areaType: 'GP',
      level: 4,
    },
  ],
  'ICB': [
    {
      code: 'E38000007',
      name: 'NHS Basildon And Brentwood ICB',
      hierarchyName: 'NHS',
      areaType: 'ICB',
      level: 2,
    },
    {
      code: 'E38000026',
      name: 'NHS Cambridgeshire and Peterborough ICB',
      hierarchyName: 'NHS',
      areaType: 'ICB',
      level: 2,
    },
    {
      code: 'E38000240',
      name: 'NHS North Central London ICB',
      hierarchyName: 'NHS',
      areaType: 'ICB',
      level: 2,
    },
    {
      code: 'E38000244',
      name: 'NHS South East London ICB',
      hierarchyName: 'NHS',
      areaType: 'ICB',
      level: 2,
    },
  ],
  'NHS region': [
    {
      code: 'E40000007',
      name: 'East of England NHS Region',
      hierarchyName: 'NHS',
      areaType: 'NHS region',
      level: 1,
    },
    {
      code: 'E40000003',
      name: 'London NHS Region',
      hierarchyName: 'NHS',
      areaType: 'NHS region',
      level: 1,
    },
    {
      code: 'E40000005',
      name: 'South East NHS Region',
      hierarchyName: 'NHS',
      areaType: 'NHS region',
      level: 1,
    },
    {
      code: 'E40000006',
      name: 'South West NHS Region',
      hierarchyName: 'NHS',
      areaType: 'NHS region',
      level: 1,
    },
    {
      code: 'E40000010',
      name: 'North West NHS Region',
      hierarchyName: 'NHS',
      areaType: 'NHS region',
      level: 1,
    },
    {
      code: 'E40000011',
      name: 'Midlands NHS Region',
      hierarchyName: 'NHS',
      areaType: 'NHS region',
      level: 1,
    },
    {
      code: 'E40000012',
      name: 'North East and Yorkshire NHS Region',
      hierarchyName: 'NHS',
      areaType: 'NHS region',
      level: 1,
    },
  ],
  'PCN': [
    {
      code: 'U15488',
      name: 'East Basildon PCN',
      hierarchyName: 'NHS',
      areaType: 'PCN',
      level: 3,
    },
    {
      code: 'U55146',
      name: 'Central Basildon PCN',
      hierarchyName: 'NHS',
      areaType: 'PCN',
      level: 3,
    },
    {
      code: 'U02795',
      name: 'North 2 Islington PCN',
      hierarchyName: 'NHS',
      areaType: 'PCN',
      level: 3,
    },
    {
      code: 'U05885',
      name: 'South Camden PCN',
      hierarchyName: 'NHS',
      areaType: 'PCN',
      level: 3,
    },
  ],
  'Regions Statistical': [
    {
      code: 'E12000001',
      name: 'North East region (statistical)',
      hierarchyName: 'Admin',
      areaType: 'Regions Statistical',
      level: 1,
    },
    {
      code: 'E12000002',
      name: 'North West region (statistical)',
      hierarchyName: 'Admin',
      areaType: 'Regions Statistical',
      level: 1,
    },
    {
      code: 'E12000003',
      name: 'Yorkshire and the Humber region (statistical)',
      hierarchyName: 'Admin',
      areaType: 'Regions Statistical',
      level: 1,
    },
  ],
};
