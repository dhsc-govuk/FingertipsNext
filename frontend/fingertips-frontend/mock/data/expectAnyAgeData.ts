import { AgeData } from '@/generated-sources/ft-api-client';

export const expectAnyAgeData = (overrides?: Partial<AgeData>) => ({
  value: expect.any(String),
  isAggregate: expect.any(Boolean),
  ...overrides,
});
