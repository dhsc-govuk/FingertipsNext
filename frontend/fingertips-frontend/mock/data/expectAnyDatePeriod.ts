import { DatePeriod } from '@/generated-sources/ft-api-client';

export const expectAnyDatePeriod = (overrides?: Partial<DatePeriod>) => ({
  type: expect.any(String),
  from: expect.any(Date),
  to: expect.any(Date),
  ...overrides,
});
