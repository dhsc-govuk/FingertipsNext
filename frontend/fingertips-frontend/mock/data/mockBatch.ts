import { Batch, BatchStatusEnum } from '@/generated-sources/ft-api-client';

export const mockBatch = (overrides?: Partial<Batch>): Batch => ({
  batchId: '41101_2020-03-07T14:22:37.123Z',
  indicatorId: 41101,
  originalFilename: 'test-file.csv',
  createdAt: new Date('2020-03-07T14:22:37.123Z'),
  publishedAt: new Date('2020-03-07T14:22:37.123Z'),
  userId: 'fd89acd7-c91f-49c0-89ab-c46d3b25b4f0',
  status: BatchStatusEnum.Received,
  ...overrides,
});
