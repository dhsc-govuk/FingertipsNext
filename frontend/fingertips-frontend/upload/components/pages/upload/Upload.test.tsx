import { IndicatorsApi } from '@/generated-sources/ft-api-client';
import { ApiClientFactory } from '@/lib/apiClient/apiClientFactory';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { mockDeep } from 'vitest-mock-extended';
import { Upload } from '.';
import { mockBatch } from '@/mock/data/mockBatch';

const mockIndicatorsApi = mockDeep<IndicatorsApi>();
ApiClientFactory.getIndicatorsApiClient = () => mockIndicatorsApi;

const apiResponsePanelTestId = 'api-response-panel';

describe('Upload page component', () => {
  it('should render the warning text', () => {
    render(<Upload batches={[mockBatch()]} />);

    expect(
      screen.getByText(
        'This is an interim tool to allow developers to demonstrate data upload to the API'
      )
    ).toBeInTheDocument();
  });

  it('should not render the API response panel before the submit button has not been clicked', () => {
    render(<Upload batches={[mockBatch()]} />);

    expect(
      screen.queryByTestId(apiResponsePanelTestId)
    ).not.toBeInTheDocument();
  });

  it('should render the API response panel when the submit button has been clicked', async () => {
    mockIndicatorsApi.indicatorsIndicatorIdDataPostRaw.mockResolvedValue({
      raw: new Response('Body', {
        status: 202,
      }),
      value: vi.fn(),
    });
    const user = userEvent.setup();

    render(<Upload batches={[mockBatch()]} />);
    await user.click(screen.getByRole('button', { name: 'Submit' }));

    expect(
      await screen.findByTestId(apiResponsePanelTestId)
    ).toBeInTheDocument();
  });

  it('should render the page heading', () => {
    render(<Upload batches={[mockBatch()]} />);

    expect(screen.getByRole('heading', { level: 1 })).toHaveTextContent(
      'Indicator data portal'
    );
  });

  it('should render the upload form', () => {
    render(<Upload batches={[mockBatch()]} />);

    expect(screen.getByLabelText(/Add indicator ID/)).toBeInTheDocument();
  });
});
