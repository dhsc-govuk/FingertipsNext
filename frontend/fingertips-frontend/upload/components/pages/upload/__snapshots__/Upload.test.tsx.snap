// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Upload page component > snapshot test 1`] = `
<DocumentFragment>
  <div
    class="src__StyledWarningText-sc-14ghfbq-0 iiTzwf sc-kAyceB jkvpBD"
  >
    <div
      class="src__IconImportantWrapper-sc-14ghfbq-1 cEUjzt"
    >
      <svg
        class="SVGBase__StyledSvg-sc-16k4s64-0 fzOiHV"
        fill="currentColor"
        height="100%"
        preserveAspectRatio="xMidYMid meet"
        version="1.1"
        viewBox="0 0 35.000000 35.000000"
        width="100%"
      >
        <title>
          icon important
        </title>
        <g
          transform="translate(0.000000,35.000000) scale(0.100000,-0.100000)"
        >
          <path
            d="M100 332 c-87 -48 -125 -155 -82 -232 48 -87 155 -125 232 -82 87 48
125 155 82 232 -48 87 -155 125 -232 82z m100 -122 c0 -53 -2 -60 -20 -60 -18
0 -20 7 -20 60 0 53 2 60 20 60 18 0 20 -7 20 -60z m0 -111 c0 -12 -7 -19 -20
-19 -19 0 -28 28 -14 43 11 11 34 -5 34 -24z"
          />
        </g>
      </svg>
    </div>
    <strong
      class="src__WarningTextWrapper-sc-14ghfbq-2 cIfUOd"
    >
      <span
        class="sc-imWYAI cvzslG"
      >
        This is an interim tool to allow developers to demonstrate data upload to the API
      </span>
    </strong>
  </div>
  <h1
    class="heading__StyledHeading-sc-1dsobun-0 eaMQId sc-jXbUNg dJyqsT"
  >
    Indicator data portal
  </h1>
  <form
    action="javascript:throw new Error('A React form was unexpectedly submitted. If you called form.submit() manually, consider using form.requestSubmit() instead. If you\\'re trying to use event.stopPropagation() in a submit event handler, consider also calling event.preventDefault().')"
  >
    <label
      class="src__Label-sc-iqzvxn-0 cjKMYe sc-aXZVg iybZny"
    >
      <span
        class="src__LabelText-sc-1lbxenh-0 bKqlAx"
      >
        <div
          class="sc-gEvEer gpmWer"
        >
          Add indicator ID
        </div>
      </span>
      <span
        class="src__HintText-sc-tq1z5r-0 PfjNt"
      >
        This is a unique identification number
      </span>
      <input
        class="src__Input-sc-1ch9crp-0 hYZdGo"
        name="indicatorId"
        type="number"
      />
    </label>
    <div
      class="src__StyledContainer-sc-1l2t4xv-0 dbrEkK sc-fqkvVR hpBnMO"
    >
      <span
        class="src__LabelText-sc-1lbxenh-0 bKqlAx"
      >
        <div
          class="sc-gEvEer gpmWer"
        >
          Add a publish date
        </div>
      </span>
      <span
        class="src__HintText-sc-tq1z5r-0 PfjNt"
      >
        The date must be today or in the future
      </span>
      <div
        class="input__StyledList-sc-una33r-1 fGlhYe"
      >
        <label
          class="src__Label-sc-iqzvxn-0 input__StyledLabel-sc-una33r-0 cjKMYe hNChCj"
        >
          <span
            class="src__LabelText-sc-1lbxenh-0 bKqlAx"
          >
            Day
          </span>
          <input
            class="src__Input-sc-1ch9crp-0 hYZdGo"
            name="publishDateDay"
            type="number"
          />
        </label>
        <label
          class="src__Label-sc-iqzvxn-0 input__StyledLabel-sc-una33r-0 cjKMYe hNChCj"
        >
          <span
            class="src__LabelText-sc-1lbxenh-0 bKqlAx"
          >
            Month
          </span>
          <input
            class="src__Input-sc-1ch9crp-0 hYZdGo"
            name="publishDateMonth"
            type="number"
          />
        </label>
        <label
          class="src__Label-sc-iqzvxn-0 input__StyledLabel-sc-una33r-0 cjKMYe gaqalV"
        >
          <span
            class="src__LabelText-sc-1lbxenh-0 bKqlAx"
          >
            Year
          </span>
          <input
            class="src__Input-sc-1ch9crp-0 hYZdGo"
            name="publishDateYear"
            type="number"
          />
        </label>
      </div>
    </div>
    <label
      class="src__Label-sc-iqzvxn-0 cjKMYe sc-dcJsrY bVOTul"
    >
      <span
        class="src__LabelText-sc-1lbxenh-0 bKqlAx"
      >
        <div
          class="sc-gEvEer gpmWer"
        >
          Upload a file
        </div>
      </span>
      <span
        class="src__HintText-sc-tq1z5r-0 PfjNt"
      >
        The file must be the CSV format provided by this service
      </span>
      <input
        accept=".csv"
        class="src__StyledInput-sc-176u3do-0 inprkn"
        name="file"
        type="file"
      />
    </label>
    <div
      class="sc-iGgWBj kVSmVC"
    >
      <div
        class="sc-gEvEer sc-eqUAAy gpmWer dRIFMJ"
      >
        Submit the uploaded file and information
      </div>
      <button
        class="src__StyledButton-sc-19ocyxv-0 eEShVH"
      >
        Submit
      </button>
    </div>
  </form>
  <div
    data-testid="batch-list-table-container"
  >
    <h4
      class="heading__StyledHeading-sc-1dsobun-0 EoZyU"
    >
      Manage upload data
    </h4>
    <table
      class="src__StyledTable-sc-16s660v-0 duOufa sc-dhKdcB egpXem"
      data-testid="batch-list-table"
    >
      <tbody
        class="TableBody-sc-1qqarm8-0"
      >
        <tr
          class="Row-sc-1y8eqfp-0"
        >
          <th
            class="Cell-sc-msv0l6-0 iftjBc sc-dAlyuH eiLaTR"
          >
            Original filename
          </th>
          <th
            class="Cell-sc-msv0l6-0 iftjBc sc-dAlyuH eiLaTR"
          >
            Timestamp
          </th>
          <th
            class="Cell-sc-msv0l6-0 iftjBc sc-dAlyuH eiLaTR"
          >
            User ID
          </th>
          <th
            class="Cell-sc-msv0l6-0 iftjBc sc-dAlyuH eiLaTR"
          >
            Publish date
          </th>
          <th
            class="Cell-sc-msv0l6-0 iftjBc sc-dAlyuH eiLaTR"
          >
            Indicator ID
          </th>
          <th
            class="Cell-sc-msv0l6-0 iftjBc sc-dAlyuH eiLaTR"
          >
            Batch ID
          </th>
          <th
            class="Cell-sc-msv0l6-0 iftjBc sc-dAlyuH eiLaTR"
          >
            Status
          </th>
          <th
            class="Cell-sc-msv0l6-0 iftjBc sc-dAlyuH eiLaTR"
          />
        </tr>
        <tr
          class="Row-sc-1y8eqfp-0"
        >
          <td
            class="Cell-sc-msv0l6-0 sc-kpDqfm gaSlHc imUseA"
          >
            test-file.csv
          </td>
          <td
            class="Cell-sc-msv0l6-0 sc-kpDqfm gaSlHc imUseA"
          >
            Sat Mar 07 2020 14:22:37 GMT+0000 (Greenwich Mean Time)
          </td>
          <td
            class="Cell-sc-msv0l6-0 sc-kpDqfm gaSlHc imUseA"
          >
            fd89acd7-c91f-49c0-89ab-c46d3b25b4f0
          </td>
          <td
            class="Cell-sc-msv0l6-0 sc-kpDqfm gaSlHc imUseA"
          >
            Sun Mar 08 2020 14:22:37 GMT+0000 (Greenwich Mean Time)
          </td>
          <td
            class="Cell-sc-msv0l6-0 sc-kpDqfm gaSlHc imUseA"
          >
            41101
          </td>
          <td
            class="Cell-sc-msv0l6-0 sc-kpDqfm gaSlHc imUseA"
          >
            41101_2020-03-07T14:22:37.123
          </td>
          <td
            class="Cell-sc-msv0l6-0 sc-kpDqfm gaSlHc imUseA"
          >
            Received
          </td>
          <td
            class="Cell-sc-msv0l6-0 sc-jlZhew gaSlHc fBsBcN"
          >
            <button
              class="src__StyledButton-sc-19ocyxv-0 eEShVH"
            >
              Delete Submission
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
